### DISCUSSION TITLE
Test cases start executing before the completion of hooks calls [before, beforeEach]

### DISCUSSION THREAD

**Discussion by vinayak-kaladhar** [pactumjs/pactum#109] [General]
Scenario: 
I have a file named **hooks.j**s with before/beforeAll() used to setup pre-requisite.

**Contents of hooks.js :**
```js
exports.mochaHooks = {
     before() {
        request.setBaseUrl(baseURL);
        request.setDefaultTimeout = 20000
        //utility.deleteExistingShifts("shifts","");
        // utility.insertSurgeryFacilityMongoDB("facilityprofiles","");
         utility.insertFacilityShiftsMongoDB("shifts","");
        // utility.insertFacilityUsersMongoDB("users","");
    }
 }
```
I have my test-scripts files also.

I trigger tests as mocha --require test/hooks.js test/{test_file.js}

But before the completion of methods in hooks.js under before annotation, test scripts start executing, resulting failure.

**Comment by tejeswarpatro18** [comment]
@vinayak-kaladhar I think this issue is more into synchronous and asynchronous function, where pactum has no part to play.
You can try making before function as async /await or else return some value from before function, So that the execution will not rush over the scripts before the hook file gets executed
